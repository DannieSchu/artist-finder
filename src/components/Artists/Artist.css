.Artist {
  width: 100%;
  padding: 1rem;
  
  th {
    color: rgb(160, 231, 243);
    background-color: rgb(22, 53, 88);
    border-radius: 50% 20% / 10% 40%;
    text-align: center;
    width: 3rem;
    height: 1.5rem;
    padding: 1rem 1rem;
  }

  td {
    color: rgb(22, 53, 88);
    text-align: center;
    width: 3rem;
    height: 1.5rem;
    padding: 1rem 1rem;
  } 
}