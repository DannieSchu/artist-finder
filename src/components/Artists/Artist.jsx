import React from 'react';
import PropTypes from 'prop-types';
import styles from '../../global-styles/item.css';

const Artist = ({ name, origin, birth, death }) => (
  <section className={styles.item}>
    <tr>
      <th>Artist</th>
      <td>{name}</td>
    </tr>
    <tr>
      {origin && <td>{origin}</td>}
      {origin && <th>Origin</th>}
    </tr>
    <tr>
      {birth && <th>DOB</th>}
      {birth && <td>{birth.slice(0, 4)}</td>}
    </tr>
    <tr>
      {death && <td>{death.slice(0, 4)}</td>}
      {death && <th>Death</th>}
    </tr>
  </section>
);

Artist.propTypes = {
  name: PropTypes.string.isRequired,
  origin: PropTypes.string,
  birth: PropTypes.string,
  death: PropTypes.string
};

export default Artist;
