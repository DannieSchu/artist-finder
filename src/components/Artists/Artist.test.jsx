import React from 'react';
import { shallow } from 'enzyme';
import Artist from './Artist.jsx';

describe('Artist', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <Artist
        name="Leonard Cohen"
        origin="Canada"
        birth="1934-09-21"
        death="1934-09-21" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
