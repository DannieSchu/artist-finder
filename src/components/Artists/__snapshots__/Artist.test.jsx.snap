// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist matches a snapshot 1`] = `
<table
  className="Artist"
>
  <tr>
    <th>
      Artist
    </th>
    <td>
      Leonard Cohen
    </td>
  </tr>
  <tr>
    <td>
      Canada
    </td>
    <th>
      Origin
    </th>
  </tr>
  <tr>
    <th>
      DOB
    </th>
    <td>
      1934
    </td>
  </tr>
  <tr>
    <td>
      1934
    </td>
    <th>
      Death
    </th>
  </tr>
</table>
`;
